lib/model.js

```diff
@@ -1655,6 +1655,15 @@ Model.dropSearchIndex = async function dropSearchIndex(name) {
   return await this.$__collection.dropSearchIndex(name);
 };
 
+
+Model.listSearchIndexes = async function listSearchIndexes() {
```

Comment:
i think there should be jsdoc so that it gets added to the documentation and also typescript types.
i dont think tests can be easily added as search indexes are atlas-exclusive (to my knowledge)